### PR TITLE
Allow capi time windows on fronts or collections, as well as editions.

### PR DESCRIPTION
**Pull request by aug24**
## What's changed?
Introduce an optional time window on the FrontTemplate and CollectionTemplate.
Use this as a hierarchy for deciding which time window to use for prefill.

## Implementation notes
This required some refactoring, and I took the opportunity to make class names consistent (eg Content* and Capi* become Capi* everywhere) and break up large functions for clarity.
I also added missing tests for the EditionsTemplating class.

### General
- [X] 🤖 Relevant tests added
- [X] ✅ CI checks / tests run locally
- [ ] 🔍 Checked on CODE

### Client
- [X] 🚫 No obvious console errors on the client (i.e. React dev mode errors)
- [X] 🎛️ No regressions with existing user interactions (i.e. all existing buttons, inputs etc. work)
- [ ] 📷 Screenshots / GIFs of relevant UI changes included

### CODE REVIEW

File: app/model/editions/templates/AmericanEdition.scala [hunk 1]
```diff
@@ -56,13 +56,11 @@ object AmericanEdition {
...
-        startOffset = 0,
-        endOffset = 0,
-        useDate = UseDateQueryParamValue.NewspaperEdition
-      )
+    timeWindowConfig = CapiTimeWindowConfigInDays(
+      startOffset = 0,
+      endOffset = 0,
     ),
+    capiDateQueryParam = CapiDateQueryParam.NewspaperEdition,
     zoneId = ZoneId.of("Europe/London"),
```
Comment (mkuzdowicz) [inline app/model/editions/templates/AmericanEdition.scala]: shouldn't `capiDateQueryParam` be part fo time window class ? it is used for time window anyway (they are used together)
I think that was the initial idea before

File: app/model/editions/templates/AustralianEdition.scala [hunk 2]
```diff
@@ -19,13 +19,11 @@ object AustralianEdition {
...
-    capiQueryPrefillParams = CapiQueryPrefillParams(
-      timeWindowConfig = CapiTimeWindowConfigInDays(
-        startOffset = -6,
-        endOffset = 0,
-        useDate = UseDateQueryParamValue.Published
-      )
+    timeWindowConfig = CapiTimeWindowConfigInDays(
+      startOffset = -6,
+      endOffset = 0,
     ),
```
Comment (mkuzdowicz) [inline app/model/editions/templates/AustralianEdition.scala]: same as comment at `AmericanEdition`

File: app/model/editions/templates/DailyEdition.scala [hunk 3]
```diff
@@ -56,13 +56,11 @@ object DailyEdition {
...
-        startOffset = 0,
-        endOffset = 0,
-        useDate = UseDateQueryParamValue.NewspaperEdition
-      )
+    timeWindowConfig = CapiTimeWindowConfigInDays(
+      startOffset = 0,
+      endOffset = 0,
     ),
+    capiDateQueryParam = CapiDateQueryParam.NewspaperEdition,
     zoneId = ZoneId.of("Europe/London"),
```
Comment (mkuzdowicz) [inline app/model/editions/templates/DailyEdition.scala]: same as comment at AmericanEdition

File: app/services/editions/EditionsTemplating.scala [hunk 4]
```diff
@@ -23,60 +23,49 @@ class EditionsTemplating(templates: PartialFunction[Edition, EditionTemplate], c
     templates.lift(edition) match {
       case Some(editionTemplate) =>
```
Comment (mkuzdowicz) [inline app/services/editions/EditionsTemplating.scala]: Good to have some refactoring here :) 👍

File: test/fixtures/TestEdition.scala [hunk 5]
```diff
@@ -18,13 +18,11 @@ object TestEdition {
...
-    capiQueryPrefillParams = CapiQueryPrefillParams(
-      timeWindowConfig = CapiTimeWindowConfigInDays(
-        startOffset = ContentQueryStartOffsetInDays,
-        endOffset = ContentQueryEndOffsetInDays,
-        useDate = UseDate,
-      )
+    timeWindowConfig = CapiTimeWindowConfigInDays(
+      startOffset = ContentQueryStartOffsetInDays,
+      endOffset = ContentQueryEndOffsetInDays,
     ),
```
Comment (mkuzdowicz) [inline test/fixtures/TestEdition.scala]: same as comment at `AmericanEdition`

File: app/model/editions/EditionsTemplates.scala [hunk 6]
```diff
@@ -179,15 +185,22 @@ trait BaseTimeWindowConfig {
 
 case class TimeWindowConfigInDays(startOffset: Int, endOffset: Int) extends BaseTimeWindowConfig
 
-case class CapiTimeWindowConfigInDays(startOffset: Int, endOffset: Int, useDate: UseDateQueryParamValue) extends BaseTimeWindowConfig
-
-case class CapiQueryPrefillParams(timeWindowConfig: CapiTimeWindowConfigInDays)
+case class CapiTimeWindowConfigInDays(startOffset: Int, endOffset: Int) extends BaseTimeWindowConfig {
+  def toCapiQueryTimeWindow(issueDate: LocalDate): CapiQueryTimeWindow = {
+    val issueDateStart = issueDate.atStartOfDay()
```
Comment (mkuzdowicz) [inline app/model/editions/EditionsTemplates.scala]: that's clever 👍

File: app/model/editions/templates/TheDummyEdition.scala [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+import model.editions.Swatch._
+import model.editions._
+import model.editions.templates.TemplateHelpers._
+
+//noinspection TypeAnnotation
+object TheDummyEdition {
+  lazy val template = EditionTemplate(
+    List(
+      FrontSpecial01 -> Daily(),
```
Comment (mkuzdowicz) [inline app/model/editions/templates/TheDummyEdition.scala]: can we have more variations in `TheDummyEdition` on how we can set time window

File: app/model/editions/templates/AmericanEdition.scala (hunk 1 above)
Comment (aug24) [reply app/model/editions/templates/AmericanEdition.scala]: After consultation with David, it had to be taken out.
useDate is specific to Issue, but time window can be specific to Issue, Front or Collection.
We don't want people to be able to set a different useDate at different points.

File: app/model/editions/templates/TheDummyEdition.scala (hunk 7 above)
Comment (aug24) [reply app/model/editions/templates/TheDummyEdition.scala]: They are present in the Training Edition - the Dummy Edition is for David to do mockups in.

**Pull request merged**